modules/javafx.graphics/src/main/java/com/sun/glass/ui/mac/MacApplication.java

```diff
@@ -455,4 +460,23 @@ public Map<String, Class<?>> getPlatformKeys() {
             Map.entry("macOS.NSColor.systemYellowColor", Color.class)
         );
     }
+
+    private static final String SUPPRESS_AWT_WARNING_PROPERTY = "javafx.preferences.suppressAppleAwtWarning";
+    private static final String AWT_APPEARANCE_PROPERTY = "apple.awt.application.appearance";
+    private boolean checkSystemAppearance = !Boolean.getBoolean(SUPPRESS_AWT_WARNING_PROPERTY);
+
+    @Override
+    public void checkPlatformPreferencesSupport() {
+        if (checkSystemAppearance && "NSApplicationAWT".equals(applicationClassName)) {
+            checkSystemAppearance = false;
+
+            if (!"system".equals(System.getProperty(AWT_APPEARANCE_PROPERTY))) {
```

Comment:
You need to wrap the call to `System.getProperty` in `doPrivileged (so you will need to create a local var).